### ISSUE TITLE
ConfigSyncer does not sync configmap/secret when new namespace is created

### ISSUE THREAD

**Issue by hugocf** [appscode/kubed#266]
### Summary

Kubed ConfigSyncer does not replicate configmaps or secrets containing the annotation `kubed.appscode.com/sync` _when a new namespace_ is created.

### Steps to Reproduce

These steps follow the [Synchronize ConfigMap tutorial][1] in a newly created Kubernetes cluster on AWS (via Kops) and highlight the issue using a configmap (although the same occurs with secrets):

[1]: https://github.com/appscode/kubed/blob/master/docs/guides/config-syncer/intra-cluster.md#synchronize-configmap

1. Prepare namespace & confimap

```shell
$ kubectl create ns demo
namespace "demo" created

$ kubectl apply -f demo-0.yaml
configmap "omni" created

$ kubectl get configmaps --all-namespaces | grep omni
demo              omni                                 2         9s

$ kubectl annotate configmap omni kubed.appscode.com/sync="" -n demo
configmap "omni" annotated

$ kubectl get configmaps --all-namespaces | grep omni
default           omni                                 2         10s
demo              omni                                 2         1m
kube-public       omni                                 2         10s
kube-system       omni                                 2         10s
```

2. Create a new namespace

```shell
$ kubectl create ns other
namespace "other" created

$ kubectl get configmaps --all-namespaces | grep omni
default           omni                                 2         1m
demo              omni                                 2         2m
kube-public       omni                                 2         1m
kube-system       omni                                 2         1m

$ kubectl get configmaps -n other
No resources found.
```

3. Change the original configmap

```shell
$ kubectl apply -f demo-1.yaml
configmap "omni" configured

$ kubectl get configmaps --all-namespaces | grep omni
default           omni                                 2         5m
demo              omni                                 2         6m
kube-public       omni                                 2         5m
kube-system       omni                                 2         5m
other             omni                                 2         17s

$ kubectl get configmaps -n other
NAME      DATA      AGE
omni      2         27m
```

### Expected Results

In the step 2 above, the `omni` configmap should have been copied to the namespace `other`.

### Actual Results

The configmap was not copied when the `other` namespace was _created_ in step 2, and was only copied after the configmap was _changed_ in step 3.

### Versions

The cluster where this issue occurs is running **Kubernetes version 1.9.7** (_with_ RBAC enabled) and we tested with the following **Kubed versions 0.5.0**, **0.6.0-rc.0**, and **master** (sha aec012666e54e7cb89483c8ede05b036cd65f56a), all with the same erroneous results.

We currently have another cluster with **Kubernetes version 1.8.4** (_without_ RBAC) and **Kubed version 0.5.0** where this issue _does not_ occur and configmaps/secrets _are copied correctly_ when a new namespace is created.

### Notes

Kubed installation was done via the command:

```shell
curl -fsSL https://raw.githubusercontent.com/appscode/kubed/$version/hack/deploy/kubed.sh | bash -s -- --rbac
```

No errors or other relevant messages appear in `kubed-operator` pod logs when the new namespace is created.

```shell
$ kubectl logs -f -n kube-system kubed-operator-7975b7cd5d-7frfv
```

After step 1:

```
...
I0515 21:28:04.198678       1 wrap.go:42] GET /healthz: (94.228µs) 200 [[kube-probe/1.9] 172.20.96.207:29274]
I0515 21:28:13.001297       1 wrap.go:42] GET /apis/kubed.appscode.com/v1alpha1: (5.571409ms) 200 [[kube-controller-manager/v1.9.7 (linux/amd64) kubernetes/dd5e1a2/system:serviceaccount:kube-system:generic-garbage-collector] 100.114.111.0:27360]
I0515 21:28:13.003839       1 wrap.go:42] GET /apis/kubed.appscode.com/v1alpha1: (2.267777ms) 200 [[kube-controller-manager/v1.9.7 (linux/amd64) kubernetes/dd5e1a2/system:serviceaccount:kube-system:resourcequota-controller] 100.114.111.0:27360]
I0515 21:28:14.200508       1 wrap.go:42] GET /healthz: (1.667793ms) 200 [[kube-probe/1.9] 172.20.96.207:29304]
I0515 21:28:22.854850       1 configmap.go:19] Creating ConfigMap default/omni.
I0515 21:28:22.871811       1 configmap.go:19] Creating ConfigMap kube-public/omni.
I0515 21:28:22.876895       1 configmap.go:19] Creating ConfigMap kube-system/omni.
I0515 21:28:24.198826       1 wrap.go:42] GET /healthz: (127.736µs) 200 [[kube-probe/1.9] 172.20.96.207:29334]
I0515 21:28:28.532300       1 wrap.go:42] GET /: (3.167786ms) 403 [[Go-http-client/2.0] 100.114.111.0:27356]
I0515 21:28:31.246270       1 wrap.go:42] GET /swagger.json: (2.046941ms) 404 [[] 100.114.111.0:27360]
...
```

After step 2 (nothing meaningful, only the regular probe-type log messages):

```
...
I0515 21:30:04.198776       1 wrap.go:42] GET /healthz: (113.326µs) 200 [[kube-probe/1.9] 172.20.96.207:29820]
I0515 21:30:13.113450       1 wrap.go:42] GET /apis/kubed.appscode.com/v1alpha1: (3.203314ms) 200 [[kube-controller-manager/v1.9.7 (linux/amd64) kubernetes/dd5e1a2/system:serviceaccount:kube-system:generic-garbage-collector] 100.114.111.0:27360]
I0515 21:30:13.114012       1 wrap.go:42] GET /apis/kubed.appscode.com/v1alpha1: (1.765069ms) 200 [[kube-controller-manager/v1.9.7 (linux/amd64) kubernetes/dd5e1a2/system:serviceaccount:kube-system:resourcequota-controller] 100.114.111.0:27360]
I0515 21:30:14.200315       1 wrap.go:42] GET /healthz: (1.720237ms) 200 [[kube-probe/1.9] 172.20.96.207:29882]
I0515 21:30:24.198711       1 wrap.go:42] GET /healthz: (94.684µs) 200 [[kube-probe/1.9] 172.20.96.207:29926]
I0515 21:30:28.523486       1 wrap.go:42] GET /: (4.810603ms) 403 [[Go-http-client/2.0] 100.114.111.0:27356]
I0515 21:30:34.201448       1 wrap.go:42] GET /healthz: (2.718323ms) 200 [[kube-probe/1.9] 172.20.96.207:29960]
I0515 21:30:43.141552       1 wrap.go:42] GET /apis/kubed.appscode.com/v1alpha1: (1.696636ms) 200 [[kube-controller-manager/v1.9.7 (linux/amd64) kubernetes/dd5e1a2/system:serviceaccount:kube-system:generic-garbage-collector] 100.114.111.0:27360]
I0515 21:30:43.142023       1 wrap.go:42] GET /apis/kubed.appscode.com/v1alpha1: (1.716618ms) 200 [[kube-controller-manager/v1.9.7 (linux/amd64) kubernetes/dd5e1a2/system:serviceaccount:kube-system:resourcequota-controller] 100.114.111.0:27360]
I0515 21:30:44.199077       1 wrap.go:42] GET /healthz: (117.05µs) 200 [[kube-probe/1.9] 172.20.96.207:29996]
I0515 21:30:54.200672       1 wrap.go:42] GET /healthz: (2.025194ms) 200 [[kube-probe/1.9] 172.20.96.207:30030]
I0515 21:30:58.522925       1 wrap.go:42] GET /: (4.568552ms) 403 [[Go-http-client/2.0] 100.114.111.0:27356]
...
```

After step 3:

```
...
I0515 21:33:44.198895       1 wrap.go:42] GET /healthz: (148.98µs) 200 [[kube-probe/1.9] 172.20.96.207:30552]
I0515 21:33:54.200549       1 wrap.go:42] GET /healthz: (1.684086ms) 200 [[kube-probe/1.9] 172.20.96.207:30582]
I0515 21:33:58.505544       1 wrap.go:42] GET /: (3.95587ms) 403 [[Go-http-client/2.0] 100.114.111.0:27356]
I0515 21:33:58.783536       1 configmap.go:52] Patching ConfigMap default/omni with {"metadata":{"annotations":{"kubed.appscode.com/origin":"{\"namespace\":\"demo\",\"name\":\"omni\",\"uid\":\"bd9f7953-5886-11e8-a4dd-069380676cb2\",\"resourceVersion\":\"976406\"}"}}}
I0515 21:33:58.803950       1 configmap.go:52] Patching ConfigMap kube-public/omni with {"metadata":{"annotations":{"kubed.appscode.com/origin":"{\"namespace\":\"demo\",\"name\":\"omni\",\"uid\":\"bd9f7953-5886-11e8-a4dd-069380676cb2\",\"resourceVersion\":\"976406\"}"}}}
I0515 21:33:58.808944       1 configmap.go:52] Patching ConfigMap kube-system/omni with {"metadata":{"annotations":{"kubed.appscode.com/origin":"{\"namespace\":\"demo\",\"name\":\"omni\",\"uid\":\"bd9f7953-5886-11e8-a4dd-069380676cb2\",\"resourceVersion\":\"976406\"}"}}}
I0515 21:33:58.813001       1 configmap.go:19] Creating ConfigMap other/omni.
I0515 21:34:04.198629       1 wrap.go:42] GET /healthz: (125.845µs) 200 [[kube-probe/1.9] 172.20.96.207:30612]
...
```

**Comment by emruz-hossain** [comment]
@hugocf Thank you for reporting. This was caused by missing "get" permission for Namespaces. Fixed in #268

**Comment by tamalsaha** [comment]
Should be fixed by https://github.com/appscode/kubed/releases/tag/0.7.0-rc.0 release. Please reopen if you are still having this issue.
